src/settings/FirestoreStorage.ts

```diff
@@ -0,0 +1,38 @@
+import { updateSettingField, getSettings } from '../services/firebase'
+import { Settings } from '../settings/index'
+
+export type FieldValue =
+    | string
+    | number
+    | Array<string>
+    | firebase.firestore.GeoPoint
+    | { [key: string]: string[] }
+
+export function persist(docId: string, settings: Settings): void {
+    updateSettingField(docId, settings)
+}
+
+export async function restore(id?: string): Promise<Settings> {
```

Comment:
Dersom `id` er undefined, eller dersom det ikkje finst database-dokument med ID-en, så vil denne funksjonen returnere `undefined`. Men kanskje den burder returnere `DEFAULT_SETTINGS` i tilfelle? Eg ser at restore i UrlStorage gjer det, i alle fall. Eg veit ikkje kva som er praktisk?

Reply:
Vi har vel egentlig skubbet denne casen litt under teppet, siden designskissene inneholder design for hva som kan skje når man besøker en URL som ikke er opprettet.

Det beste inntil videre er kanskje å vise en 404-skjerm eller lignende når settings ikke kan lastes inn. Er det innafor å bare redirecte til forsiden i fra `useSettings` inntil videre, eller blir det for billig en løsning synes du?

Reply:
Redirect til forsida er ein grei start, det. Kva skjer no dersom ID er undefinert? Kræsjer sida?

Reply:
Ja 🙃

Reply:
Ok, me bør håndtere det før prodsetting, men det kan skje i ny PR om de vil.